server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts

```diff
@@ -304,10 +304,27 @@ export default class ServiceProviderReferralsController {
       ActionPlanUtils.getLatestApprovedActionPlanSummary(approvedActionPlanSummaries)
     let actionPlanAppointments: ActionPlanAppointment[] = []
     if (latestApprovedActionPlanSummary !== null) {
-      actionPlanAppointments = await this.interventionsService.getActionPlanAppointments(
-        accessToken,
-        latestApprovedActionPlanSummary.id
-      )
+      actionPlanAppointments = await this.interventionsService
+        .getActionPlanAppointments(accessToken, latestApprovedActionPlanSummary.id)
+        .then(actionPlanAppointmentsReturned => {
+          return actionPlanAppointmentsReturned
+            .flatMap(x => {
+              return [
+                x,
+                x.oldAppointments
```

Comment:
would it work if you changed this to
```
...(x.oldAppointments
   ? { map statement}
: []),
```
then you could remove the following flat() and filter() lines?

Reply:
this doesnt seem to work

Reply:
changing the code to
```
    if (latestApprovedActionPlanSummary !== null) {
      actionPlanAppointments = await this.interventionsService
        .getActionPlanAppointments(accessToken, latestApprovedActionPlanSummary.id)
        .then(actionPlanAppointmentsReturned => {
          return actionPlanAppointmentsReturned.flatMap(x => {
            return [
              x,
              ...(x.oldAppointments
                ? x.oldAppointments!.map(y => {
                    const actionPlanAppointment: ActionPlanAppointment = {
                      sessionNumber: x.sessionNumber,
                      ...y,
                    }
                    return actionPlanAppointment
                  })
                : []),
            ]
          })
        })
    }
```

seems to work fine for me locally, the spread operator drops undefined/empty array entries so should do the same as having
```
            .flat()
            .filter(x => x)
```